conans/test/unittests/tools/cmake/test_cmaketoolchain.py

```diff
@@ -223,6 +223,19 @@ def test_toolset(conanfile_msvc):
     assert 'CMAKE_CXX_STANDARD 20' in toolchain.content
 
 
+def test_toolset_update_version(conanfile_msvc):
+    conanfile_msvc.settings.compiler.update = "8"
+    toolchain = CMakeToolchain(conanfile_msvc)
+    assert 'set(CMAKE_GENERATOR_TOOLSET "version=14.38" CACHE STRING "" FORCE)' in toolchain.content
+
+
+def test_toolset_update_version_overflow(conanfile_msvc):
+    # https://github.com/conan-io/conan/issues/15583
+    conanfile_msvc.settings.compiler.update = "10"
```

Comment:
Good catch, I didn't see it, because I was confused with the ``v143`` toolset version.

I don't love that now the ``compiler.version`` would not longer have a corresponding IDE version matching 😞